graylog-storage-elasticsearch6/src/main/java/org/graylog/storage/elasticsearch6/IndicesAdapterES6.java

```diff
@@ -362,25 +362,35 @@ public IndexRangeStats indexRangeStatsOfIndex(String index) {
 
     @Override
     public HealthStatus waitForRecovery(String index) {
-        final Health.Status status = waitForStatus(index, Health.Status.YELLOW);
+        return waitForRecovery(index, 30);
+    }
+
+    @Override
+    public HealthStatus waitForRecovery(String index, int timeout) {
+        final Health.Status status = waitForStatus(index, Health.Status.YELLOW, timeout);
         return mapHealthStatus(status);
     }
 
     private HealthStatus mapHealthStatus(Health.Status status) {
         return HealthStatus.fromString(status.toString());
     }
 
-    private Health.Status waitForStatus(String index, @SuppressWarnings("SameParameterValue") Health.Status clusterHealthStatus) {
+    private Health.Status waitForStatus(String index, @SuppressWarnings("SameParameterValue") Health.Status clusterHealthStatus, int timeout) {
         final Health request = new Health.Builder()
                 .addIndex(index)
                 .waitForStatus(clusterHealthStatus)
+                .timeout(timeout)
                 .build();
-        final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read health status for index " + index);
+        final JestResult jestResult = JestUtils.executeUnsafe(jestClient, null, request, () -> "Couldn't read health status for index " + index);
```

Comment:
Hmm, I think this is an issue because most consumers of `Indices#waitForRecovery(String)` expect the method to throw an error and don't use the return value. We would need to adjust all consumers to check the return value and abort or retry their execution flow if an index is red or the timeout kicks in. 

![image](https://user-images.githubusercontent.com/461/106017659-f8421600-60c0-11eb-959b-4210e39b2e25.png)